src/services/dashboards/config.ts

```diff
@@ -18,3 +20,55 @@ export const GRANULARITY = {
     MONTHLY: 'MONTHLY',
     YEARLY: 'YEARLY',
 } as const;
+
+export const VARIABLE_SELECTION_TYPES = ['SINGLE', 'MULTI'] as const;
+type VariableSelectionType = typeof VARIABLE_SELECTION_TYPES[number];
+
+export const VARIABLE_TYPES = ['MANAGED', 'CUSTOM'] as const;
+type VariableType = typeof VARIABLE_TYPES[number];
+
+export type DashboardScope = keyof typeof DASHBOARD_SCOPE;
+export type DashboardViewer = keyof typeof DASHBOARD_VIEWER;
+
+export interface DateRange {
+    start?: string;
+    end?: string;
+}
+
+type DashboardVariables = Record<string, any|any[]>;
+
+interface DashboardSettings {
+    date_range: {
+        enabled: boolean;
+    } & DateRange;
+    currency: {
+        enabled: boolean;
+        value?: string;
+    }
+}
+
+export interface DashboardConfig {
+    name: string;
+    layouts: DashboardLayoutWidgetInfo[][];
+    dashboard_variables: DashboardVariables;
+    settings: DashboardSettings;
+    dashboard_variables_schema: DashboardVariablesSchema;
+    labels: string[];
+    version: string;
+}
+
+// variables schema
+export interface DashboardVariableSchemaProperty {
+    variable_type: VariableType;
+    use?: boolean;
+    selection_type?: VariableSelectionType;
+    options?: any[];
+    name?: string; // for managed variables
+    default_use?: boolean; // for managed variables. If property exists, ignore it. Otherwise, replace value of 'use'.
+}
+export interface DashboardVariablesSchema {
+    properties: {
+        [key: string]: DashboardVariableSchemaProperty;
+    };
+    order: string[];
+}
```

Comment:
description: This is added type for dashboard variables schema.

We discussed to make properties in an array, but I finally chose an object.
When managed as an array, the more values there are in the schema, the more difficult it is to see the order at a glance.
Debugging will be equally difficult. The dev tool implicitly shows the objects in the array, but to check the order, we will have to expand and check each object one by one.
This seemed too inefficient, so I changed to managing the `order` separately.